### ISSUE TITLE
Possible use of hipBLAS for AMD graphic card ?

### ISSUE THREAD

**Issue by slhad** [occ-ai/obs-localvocal#143]
TLDR : Hello, Is it possible to have a hipBLAS version in // to CLBLAS ?

After testing obs-localvocal on Windows 11 with a AMD RX 7900XTX, I found out that the GPU is barely used and almost everything run on CPU and computer memory.

I have seen that you use CLBlast that have good general compatibility but would it be possible to make a version using hipBLAS ? -> https://github.com/ROCm/hipBLAS which use ROCm [support rocBLAS and cuBLAS](https://rocm.docs.amd.com/projects/hipBLAS/en/latest/) and seem to use a lot more the GPU and less CPU+ram

I ran some tests on a WSL2 Ubuntu 22.04 instance configured to run ComfyUI with ROCm and pytorch+torch+torch audio+torch vision (whisper and huggingface python libs were installed as a dependencies) (performance are better than ZLUDA+ROCm , yes even in WSL)

and used the AMD tutorial at https://rocm.blogs.amd.com/artificial-intelligence/whisper/README.html , only made the 2 python files, downloaded a voice generated sample from "listen" of Bing+Copilot speech

```
slash@DESKTOP:~/whisper$ python3 huggingfacetransformer.py
/home/slash/.local/lib/python3.10/site-packages/huggingface_hub/file_download.py:1132: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
  warnings.warn(
/home/slash/.local/lib/python3.10/site-packages/torch/nn/modules/linear.py:125: UserWarning: Attempting to use hipBLASLt on an unsupported architecture! Overriding blas backend to hipblas (Triggered internally at ../aten/src/ATen/Context.cpp:288.)
  return F.linear(input, self.weight, self.bias)
 Of course, I can provide an example, and the passion to make a difference. So let's continue to dream, work, and make a difference. Together, we can build a future we can all be proud of. Thank you all. This speech will take about a minute to deliver at a normal pace. Feel free to modify it as per your needs. To convert it into a WAV file, you can use text-to-speech software available online or on your computer. Let me know if you need help with anything else.

slash@DESKTOP:~/whisper$ python3 whispertest.py
/home/slash/.local/lib/python3.10/site-packages/whisper/__init__.py:146: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  checkpoint = torch.load(fp, map_location=device)
/home/slash/.local/lib/python3.10/site-packages/torch/cuda/__init__.py:698: UserWarning: Can't initialize amdsmi - Error code: 34
  warnings.warn(f"Can't initialize amdsmi - Error code: {e.err_code}")
/home/slash/.local/lib/python3.10/site-packages/whisper/model.py:37: UserWarning: Attempting to use hipBLASLt on an unsupported architecture! Overriding blas backend to hipblas (Triggered internally at ../aten/src/ATen/Context.cpp:288.)
  return F.linear(
 Of course, I can provide an example of a one-minute speech in English. Dear friends, it's a pleasure to stand before you today. We are gathered here to celebrate not just our achievements, but also our aspirations. We've come a long way, but the journey is just beginning. Each day offers us a new opportunity to learn, grow, and improve. I am convinced that our future is bright, for it is shaped by each one of us. We have the courage to dream big, the determination to work hard, and the passion to make a difference. So let's continue to dream, work, and make a difference. Together, we can build a future we can all be proud of. Thank you all. This speech will take about a minute to deliver at a normal pace. Feel free to modify it as per your needs. To convert it into a WAV file, you can use text-to-speech software available online or on your computer. Let me know if you need help with anything else.
```

I know that theses tests are using python but the lib hipBLAS is made of c/c++ like CLBlast , I think there is a real performance gain possible for AMD cards (or anything using rocBLAS or cuBLAS) compared to CLBast

**Comment by royshil** [comment]
i think it's possible

https://github.com/ggerganov/whisper.cpp/blob/master/ggml/CMakeLists.txt#L126

recenly GGML has added hipBLAS support

I'm in the process of building whisper.cpp again for localvocal https://github.com/occ-ai/occ-ai-dep-whispercpp/pull/3
so i will research hipBLAS as well

**Comment by royshil** [comment]
this is in the works and will be available in the next version of LV
https://github.com/occ-ai/occ-ai-dep-whispercpp/pull/3

**Comment by royshil** [comment]
released in v0.3.4

**Comment by Ineedhelpplx** [comment]
I'm having this same problem with GPU : AMD Radeon RX 9070 XT !! T_T 
Localvocal only using CPU and not touching GPU at all. Is this fix possible for me too somehow??
